sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/query/ContinuationResumeLogicTests.java

```diff
@@ -0,0 +1,237 @@
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ContinuationResumeLogicTests {
+
+    private static void validateInitializationInfo(
+        Map<FeedRangeEpkImpl, IPartitionedToken> expectedLeftMapping,
+        Map<FeedRangeEpkImpl, IPartitionedToken> expectedTargetMapping,
+        Map<FeedRangeEpkImpl, IPartitionedToken> expectedRightMapping,
+        List<FeedRangeEpkImpl> partitionKeyRanges,
+        List<IPartitionedToken> partitionedTokens) {
+
+        PartitionMapper.PartitionMapping<IPartitionedToken> partitionMapping =
+            PartitionMapper.getPartitionMapping(partitionKeyRanges,
+                                                partitionedTokens);
+
+        assertThat(expectedLeftMapping).isEqualTo(partitionMapping.getMappingLeftOfTarget());
+        assertThat(expectedTargetMapping).isEqualTo(partitionMapping.getTargetMapping());
+        assertThat(expectedRightMapping).isEqualTo(partitionMapping.getMappingRightOfTarget());
+    }
+
+    private static FeedRangeEpkImpl combineRanges(FeedRangeEpkImpl range1, FeedRangeEpkImpl range2) {
+        assert (range1 != null);
+        assert (range2 != null);
+        assertThat(range1.getRange().getMin().compareTo(range2.getRange().getMin())).isLessThan(0);
+        assertThat(range1.getRange().getMax()).isEqualTo(range2.getRange().getMin());
+
+        return new FeedRangeEpkImpl(new Range<>(range1.getRange().getMin(),
+                                                range2.getRange().getMax(),
+                                                true,
+                                                false));
+    }
+
+    private static Map<FeedRangeEpkImpl, IPartitionedToken> mapping(
+        FeedRangeEpkImpl feedRangeEpk,
+        IPartitionedToken token) {
+
+        Map<FeedRangeEpkImpl, IPartitionedToken> mapping = new HashMap<>();
+        mapping.put(feedRangeEpk, token);
+        return mapping;
+    }
+
+    private static Range<String> createRange(String min, String max) {
+        return new Range<>(min, max, true, false);
+    }
+
+    @Test(groups = {"unit"})
+    public void resumeEmptyStart() {
+        Range<String> range = createRange(Strings.Emtpy, "A");
+        FeedRangeEpkImpl range1 = new FeedRangeEpkImpl(range);
+        FeedRangeEpkImpl range2 = new FeedRangeEpkImpl(createRange("A", "B"));
+        FeedRangeEpkImpl range3 = new FeedRangeEpkImpl(createRange("B", Strings.Emtpy));
+        IPartitionedToken token = new CompositeContinuationToken(UUID.randomUUID().toString(),
+                                                                 range);
+
+        validateInitializationInfo(new HashMap<>(),
+                                   mapping(range1, token),
+                                   mapping(combineRanges(range2, range3), null),
```

Comment:
>null [](start = 74, length = 4)

annotate literals.

Reply:
intellij does that automatically. It looks a bit cluttered with double annotation